pkg/controller/controller_multiplex.go

```diff
@@ -0,0 +1,175 @@
+/*
```

Comment:
What's the thought behind the name of this file?

Reply:
a multiplexer takes an input and figures out where it should be output, a one to many relationship was the thought. So unknown object in, figure out it's type and operate on for the concrete type.